apps/angular/src/app/features/anime/table-view/table-view.component.ts

```diff
@@ -1,32 +1,132 @@
+  field: string;
+
+  /** Name of filter. */
+  title: string;
+
+  /** Is a filter selected. */
+  isSelect: boolean;
+}
+
+const DEFAULT_SORT_FIELD = 'title_eng';
+const DEFAULT_SORT_DIRECTION = 'asc';
+
 /** Table view component. */
 @Component({
   selector: 'app-table-view',
   templateUrl: './table-view.component.html',
   styleUrls: ['./table-view.component.css'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  encapsulation: ViewEncapsulation.None,
 })
 export class TableViewComponent {
 
+  /** Total number of records for the current query. */
+  public animeListCount = 0;
+
+  /** Number of records per page. */
+  public pageSize = 5;
+
+  /** All possible type filters. */
+  public filterListByType: FilterItem[] = [];
+
+  /** Value of search input. */
+  public readonly search = new FormControl('');
+
+  /** Filter by type. */
+  public readonly typeFilter = new FormControl<string[]>(['TV']);
+
+  /** Current sort settings. */
+  public readonly sort$ = new BehaviorSubject<TableSort>({
+    field: DEFAULT_SORT_FIELD,
+    direction: DEFAULT_SORT_DIRECTION,
+  });
+
+  /** Current page number. */
+  public readonly currentPageNumber$ = new BehaviorSubject<number>(0);
+
   /** Table columns names. */
   public readonly displayedColumns: readonly string[] = ['image', 'title-english', 'title-japanese', 'aired-start', 'type', 'status'];
 
   /** Anime list. */
-  public readonly animeList$: Observable<readonly AnimeBase[]>;
+  public readonly animeList$: Observable<readonly AnimeBase[]> | undefined;
 
   public constructor(private readonly animeService: AnimeService) {
-    this.animeList$ = this.getAnimeList();
-  }
+    this.setFilterListByType();
+    this.setPageSize();
+    this.setInputValues();
 
-  private getAnimeList(): Observable<readonly AnimeBase[]> {
-    return this.animeService.fetchAnimeList().pipe(
-      map(pagination => pagination.results),
+    const params$ = this.currentPageNumber$.pipe(
+      combineLatestWith(
+        this.search.valueChanges.pipe(
+          startWith(this.search.value),
+          tap(() => this.currentPageNumber$.next(0)),
+        ),
+        this.typeFilter.valueChanges.pipe(
+          startWith(this.typeFilter.value),
+          tap(() => this.currentPageNumber$.next(0)),
+        ),
+        this.sort$,
+      ),
+      debounceTime(500),
     );
+
+    this.animeList$ = params$.pipe(
+      switchMap(([pageNumber, search, typeFilter, sort]) => this.animeService.fetchAnimeList({
+        pageNumber,
+        sort,
+        filter: { byType: typeFilter !== null ? typeFilter : ['TV'] },
+        search: search !== null ? search : '',
+        limit: this.pageSize,
+      })),
+      map(animeList => {
+        this.animeListCount = animeList.count;
+        return animeList.results;
+      }),
+      tap(() => this.goToTop()),
+    );
+  }
+
+  /**
+   * Handlers pagination change.
+   * @param event Paginator event.
+   */
+  public handlePaginationChange(event: PageEvent): void {
```

Comment:
On Angular projects, we are using `on` prefix for handle methods of components
```suggestion
  public onPaginationChange(event: PageEvent): void {
```